src/utils/parsingUtils.ts

```diff
@@ -451,3 +460,40 @@ export function parseLanguage(data: any): ILanguage {
   validate(data, "ILanguage");
   return data;
 }
+
+export function parseEvent(item: any): IEvent[] {
+  const { content, id, featured_media, occasions, slug, title } = item;
+  // If given a Date in the past, occasions are empty in the API
+  if (!occasions) {
+    return [];
+  }
+  const events: IEvent[] = [];
+  const baseEvent: any = {
+    description: content.plain_text,
+    eventId: Number(id),
+    imageUrl: featured_media.source_url,
+    name: title.plain_text,
+    slug,
+  };
+  occasions.forEach((occasion: any) => {
+    const event: IEvent = { ...baseEvent };
+    event.dateStart = new Date(occasion.start_date);
+    event.dateEnd = new Date(occasion.end_date);
+    event.id = hash(id + event.dateStart.toISOString());
+
+    try {
+      if (occasion.location) {
+        event.location = parseLocation(occasion.location);
+      } else {
+        event.location = parseLocation(item.location);
```

Comment:
Maybe shorten to `event.location = parseLocation(occason.location || item.location);`